src/api/overlays/components/OverlayComponent.vue

```
@@ -140,8 +140,10 @@
                 }
             },
             buttonClickHandler: function (method) {
-                method();
-                this.destroy();
+                if (method && typeof method === 'function') {
```

Are there scenarios where this check is necessary?

---

Trying to avoid breaking errors, if devs forgot to pass in a method as a function while creating overlays

---

I think generally it's preferable to fail fast in these cases. An error with a stack trace in the log should be quite apparent to a developer. If we protect against that with an `if` statement the function does nothing, and the mistake is less obvious. The dev will spend time digging around trying to figure out why something hasn't worked.